src/components/common/Autocomplete/CurrencyAutocomplete/CurrencyAutocomplete.tsx

```diff
@@ -0,0 +1,52 @@
+import { Autocomplete, AutocompleteProps, TextField } from '@mui/material';
+import { useApiConstants } from 'src/components/Constants/UseApiConstants';
+
+interface CurrencyAutocompleteProps
+  extends Partial<AutocompleteProps<any, boolean, boolean, boolean>> {
+  textFieldLabel?: string;
+  textFieldPlaceholder?: string;
+  textFieldStyles?: React.CSSProperties;
+  textFieldAutoFocus?: boolean;
+  textFieldError?: boolean;
```

Comment:
Instead of having multiple props, can we just have one `textFieldProps`? That way in the future if we want to customize one of the props, we don't have to add a new one to this component.

If you want `autoFocus` to default to `true` like it is now, we can do `renderInput={(params) => <TextField {...params} autoFocus {...textFieldProps} />}`. Then users can override if if they want by setting `textFieldProps={{ autoFocus: false }}`.

```suggestion
  textFieldProps?: Partial<TextFieldProps>;
```